pkg/pool-manager/pod_pool.go

```diff
@@ -173,12 +176,15 @@ func (p *PoolManager) allocatePodFromPool(network *multus.NetworkSelectionElemen
 		// the pod name may have not been updated in the webhook context yet,
 		// so we use a temp pod name and update it the the controller
 		podFullName = tempPodName
-		p.macPoolMap.createOrUpdateEntry(macAddr.String(), podFullName, network.Name)
+		if isNotDryRun {
```

Comment:
i can move isNotDryRun also as a parameter to createOrUpdateEntry which will return without doing anything, if needed